include/gradual_flow/FlowLimitedPath.h

```diff
@@ -0,0 +1,452 @@
+// Copyright (c) 2024 UltiMaker
+// CuraEngine is released under the terms of the AGPLv3 or higher
+
+#ifndef GRADUAL_FLOW_GCODE_PATH_H
+#define GRADUAL_FLOW_GCODE_PATH_H
+
+#include <optional>
+#include <vector>
+
+#include <fmt/format.h>
+#include <range/v3/all.hpp>
+#include <range/v3/range/conversion.hpp>
+#include <range/v3/view/take.hpp>
+#include <spdlog/spdlog.h>
+
+#include "gradual_flow/PointContainer.h"
+#include "gradual_flow/Utils.h"
+#include "pathPlanning/GCodePath.h"
+
+namespace cura::gradual_flow
+{
+
+enum class FlowState
+{
+    STABLE,
+    TRANSITION,
+    UNDEFINED
+};
+
+struct FlowLimitedPath
+{
+    const GCodePath* original_gcode_path_data;
+    geometry::Polyline<> points;
```

Comment:
Can't we use CuraEngine's path data type here. It made sense to introduce a new path data type in the GradualFlow repo since there was no path data struct defined there yet. But in the curaengine repo we could just re-use the existing one. The whole `PointContainer` class is a bit redundant here.

Reply:
Well, yes we could, and we should, but I'm afraid that will break a lot of things, and be a much bigger work... I may be wrong though, so I'll give it a try and see where it takes me.

Reply:
Done, it turned out to be very easy, because the geometry concepts were actually used exclusively as points sets.